overlord/hookstate/hookmgr.go

```diff
@@ -143,14 +143,37 @@ func (m *HookManager) Stop() {
 	m.runner.Stop()
 }
 
+func (m *HookManager) snapContext(contextID string) (context *Context, err error) {
```

Comment:
s/snap/ephemeral/, will help us understanding the flow better once we forget about it.